esp-hal/src/peripheral.rs

```diff
@@ -257,9 +257,14 @@ mod peripheral_macros {
                 ),* $(,)?
             ]
         ) => {
+            use core::sync::atomic::Ordering;
+
 
             /// Contains the generated peripherals which implement [`Peripheral`]
             mod peripherals {
+                use core::sync::atomic::AtomicBool;
+                pub(crate) static INITIALIZED: AtomicBool = AtomicBool::new(true);
```

Comment:
This should be initialized to `false` and you should use `portable_atomic::AtomicBool` otherwise this won't work on a few of the devices.